ingestion/src/metadata/profiler/api/workflow.py

```diff
@@ -428,68 +335,81 @@ def copy_service_config(self, database) -> DatabaseService.__config__:
-                            service_connection_config=copied_service_config,
-                            table_entity=entity,
-                        )
-                        self.create_profiler_obj(entity, profiler_interface)
-                        profile: ProfilerResponse = self.profiler_obj.process(
-                            self.source_config.generateSampleData
-                        )
-                        profiler_interface.close()
-                        if hasattr(self, "sink"):
-                            self.sink.write_record(profile)
-                        self.status.failures.extend(
-                            profiler_interface.processor_status.failures
-                        )  # we can have column level failures we need to report
-                        self.status.processed(entity.fullyQualifiedName.__root__)  # type: ignore
-                        self.source_status.scanned(entity.fullyQualifiedName.__root__)  # type: ignore
-                    except Exception as exc:  # pylint: disable=broad-except
-
-                        logger.debug(traceback.format_exc())
-                        logger.warning(
-                            "Unexpected exception processing entity "
-                            f"[{entity.fullyQualifiedName.__root__}]: {exc}"  # type: ignore
-                        )
-                        self.status.failures.extend(
-                            profiler_interface.processor_status.failures  # type: ignore
-                        )
-                        self.source_status.failure(
-                            entity.fullyQualifiedName.__root__, f"{exc}"  # type: ignore
-                        )
-            except Exception as exc:  # pylint: disable=broad-except
-                logger.debug(traceback.format_exc())
-                logger.error(
-                    f"Unexpected exception executing in database [{database}]: {exc}"
+                # if we fail to instatiate a profiler_interface, we won't have a profiler_interface variable
+                self.status.failures.extend(
+                    profiler_interface.processor_status.failures  # type: ignore
                 )
+            except UnboundLocalError:
+                pass
+            self.source_status.failure(
+                entity.fullyQualifiedName.__root__, f"{exc}"  # type: ignore
+            )
+            profile = None  # type: ignore
+        else:
+            return profile, profiler_interface.processor_status.failures
+
+        return None, None
 
     def execute(self):
         """
         Run the profiling and tests
         """
 
         try:
-            self.run_profiler_workflow()
+            for database in self.get_database_entities():
+                copied_service_config = self.copy_service_config(database)
+                sqa_metadata = (
+                    MetaData()
+                    if not isinstance(copied_service_config, DatalakeConnection)
```

Comment:
nit - if we start adding instance validations, should we keep it a tad more generic to allow room for growth? For example, for when we add Spark, or any other source to profile

Reply:
Do you have an example in mind?

Reply:
Instead of `isinstance(copied_service_config, DatalakeConnection)`, we could have `isinstace(copied_service_config,  SUPPORTED_NON_SQA_CONNECTIONS` or something like that, where `SUPPORTED_NON_SQA_CONNECTIONS = (DatalakeConnection, DeltalakeConnection,....)`.

I believe if we pass a tuple of types this should work https://stackoverflow.com/questions/33311258/python-check-if-variable-isinstance-of-any-type-in-list

Reply:
then whenever we add a new profiler, we just need to prep the implementation of the metrics etc, and add the class in `SUPPORTED_NON_SQA_CONNECTIONS` without having to change any other code

Reply:
Makes sense. I'll update this.